demos/TicTacToe/Classes/util.h

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef TICTACTOE_DEMO_CLASSES_UTIL_H_
+#define TICTACTOE_DEMO_CLASSES_UTIL_H_
+
+#include <random>
```

Comment:
Please go through these headers and determine those which are required for this header and those that need to be in the cpp file only, pruning those that are unnecessary.  For instance I don't *think* that you need random.h or cocos2d.h, or most of the firebase headers (except the one that defines the futures).

Reply:
The reason that we do this is to reduce compiler times (along with other reasons).  Any file that includes this header will require the at the toolchain append all of these other header files. But if they're not needed here then that's adding cost in time for no true reason.